pkg/server/api/entry/v1/service.go

```diff
@@ -168,24 +168,32 @@ func (s *Service) createEntry(ctx context.Context, e *types.Entry, outputMask *t
 
 	log = log.WithField(telemetry.SPIFFEID, cEntry.SpiffeId)
 
-	// Validates that there is no similar entry
-	if isUniqueStatus := s.isEntryUnique(ctx, cEntry, log); isUniqueStatus != nil {
+	existingEntry, err := s.getExistingEntry(ctx, cEntry)
+	if err != nil {
 		return &entry.BatchCreateEntryResponse_Result{
-			Status: isUniqueStatus,
+			Status: api.MakeStatus(log, codes.Internal, "failed to list entries", err),
 		}
 	}
 
-	// Create entry
-	resp, err := s.ds.CreateRegistrationEntry(ctx, &datastore.CreateRegistrationEntryRequest{
-		Entry: cEntry,
-	})
-	if err != nil {
-		return &entry.BatchCreateEntryResponse_Result{
-			Status: api.MakeStatus(log, codes.Internal, "failed to create entry", err),
+	resultStatus := api.OK()
+	regEntry := existingEntry
+
+	if existingEntry == nil {
+		// Create entry
+		resp, err := s.ds.CreateRegistrationEntry(ctx, &datastore.CreateRegistrationEntryRequest{
+			Entry: cEntry,
+		})
+		if err != nil {
+			return &entry.BatchCreateEntryResponse_Result{
+				Status: api.MakeStatus(log, codes.Internal, "failed to create entry", err),
+			}
 		}
+		regEntry = resp.Entry
+	} else {
+		resultStatus = api.CreateStatus(codes.AlreadyExists, codes.AlreadyExists.String())
```

Comment:
Hmm... "entry already exists" was already a little vague. "already exists" will be even more vague. Maybe we can take this opportunity to get a little more specific about what kind of entry already existed?
```suggestion
		resultStatus = api.CreateStatus(codes.AlreadyExists, "similar entry already exists")
```